web/src/routes/RepositoryDetails/Tags/TagsActions.tsx

```diff
@@ -57,8 +65,29 @@ export default function TagActions(props: TagActionsProps) {
     >
       Change expiration
     </DropdownItem>,
+    <DropdownItem
+      key="mutable"
+      onClick={() => {
+        setIsOpen(false);
+        setImmutableModalOptions((prevOptions) => ({
+          ...prevOptions,
+          isOpen: !prevOptions.isOpen,
+          immutable: props.tags.some((tag: Tag) => tag.immutable)
+            ? false
+            : true,
+        }));
+      }}
+    >
+      {props.tags.some((tag: Tag) => tag.immutable)
+        ? 'Set mutable'
+        : 'Set immutable'}
+    </DropdownItem>,
+  ];
+
+  dropdownItems.push(
     <DropdownItem
       key="delete-tag-action"
+      isDisabled={props.tags.some((tag: Tag) => tag.immutable)}
```

Comment:
This should be added to the edit labels and edit expiration options too

Reply:
Great catch, updated.